### PR TITLE
MDS-347 | Added v2 endpoint implementation

### PR DESCRIPTION
**Pull request by owen-opencast**

### CODE REVIEW

**Review by parkeradam — APPROVED** [review]

File: app/uk/gov/hmrc/individualsmatchingapi/controllers/CommonController.scala [hunk 1]
```diff
@@ -73,18 +73,17 @@ trait PrivilegedAuthentication extends AuthorisedFunctions {
   def authPredicate(scopes: Iterable[String]): Predicate =
     scopes.map(Enrolment(_): Predicate).reduce(_ and _)
 
-  def requiresPrivilegedAuthentication(endpointScopes: Iterable[String])(
-    implicit hc: HeaderCarrier): Future[List[String]] = {
+  def requiresPrivilegedAuthentication(endpointScopes: Iterable[String])(f: Iterable[String] => Future[Result])(
```
Comment (ajchilts) [inline app/uk/gov/hmrc/individualsmatchingapi/controllers/CommonController.scala:76]: Should the common controller be in a V2 namespace to ensure it is never pulled into V1?

File: app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala [hunk 2]
```diff
@@ -18,33 +18,42 @@ package uk.gov.hmrc.individualsmatchingapi.controllers.v2
...
-      .recover(recovery)
+    requiresPrivilegedAuthentication(scopeService.getAllScopes) { _ =>
+      withJsonBody[CitizenMatchingRequest] { matchCitizen =>
+        citizenMatchingService.matchCitizen(matchCitizen) map { matchId =>
+          val selfLink = HalLink("self", s"/individuals/matching/")
+          val individualLink = HalLink(
+            "individual",
+            s"/individuals/matching/$matchId",
+            name = Option("GET"),
+            title = Option("Individual Details"))
```
Comment (ajchilts) [inline app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala:52]: Could this cause confusion with the Individuals Details API? Do we need to rethink this title? Maybe one for Lisa?

File: test/component/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingControllerSpec.scala [hunk 3]
```diff
@@ -54,17 +66,31 @@ class PrivilegedCitizenMatchingControllerSpec extends BaseSpec {
...
       And("The response contains a valid payload")
-      response.body shouldBe "{\"statusCode\":500,\"message\":\"NOT_IMPLEMENTED\"}"
+      parse(response.body) shouldBe parse(
+        s"""
+           {
+             "_links": {
+               "individual": {
+                 "href": "/individuals/matching/$sandboxMatchId",
+                 "name": "GET",
+                 "title": "Individual Details"
```
Comment (ajchilts) [inline test/component/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingControllerSpec.scala:80]: Title may change to something less conflicting with Individuals Details after a chat with Lisa.

**Review by ajchilts — CHANGES_REQUESTED** [review]
Nothing major in the comments. Looks great :-) 

One potential change if we think it is valid.

**Review by ajchilts — APPROVED** [review]

**Pull request merged**